core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,805 @@ def update_question_suggestion(
+            stat.accepted_questions_without_reviewer_edits_count)
+        stats_model.first_contribution_date = stat.first_contribution_date
+        stats_model.last_contribution_date = stat.last_contribution_date
+
+    suggestion_models.QuestionContributionStatsModel.update_timestamps_multi(  # type: ignore[type-var]
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionContributionStatsModel.put_multi(  # type: ignore[type-var]
+        stats_models_to_update)
+
+
+def _update_question_review_stats_models(
+    question_review_stats: List[
+        suggestion_registry.QuestionReviewStats]) -> None:
+    """Updates QuestionReviewStatsModel models for given question
+    review stats.
+
+    Args:
+        question_review_stats: list(QuestionReviewStats). A list of
+            QuestionReviewStats domain objects.
+    """
+    stats_ids = []
+    stats_dict = {}
+
+    for stat in question_review_stats:
+        stat_id = suggestion_models.QuestionReviewStatsModel.build_id(
+            stat.contributor_user_id, stat.topic_id)
+        stats_ids.append(stat_id)
+        stats_dict[stat_id] = stat
+
+    stats_models_to_update = (
+        suggestion_models.QuestionReviewStatsModel.get_multi(stats_ids))
+
+    for stats_model in stats_models_to_update:
+
+         # Ruling out the possibility of None for mypy type checking.
+        assert stats_model is not None
+        stat = stats_dict[stats_model.id]
+        stats_model.reviewed_questions_count = (
+            stat.reviewed_questions_count)
+        stats_model.accepted_questions_count = (
+            stat.accepted_questions_count)
+        stats_model.accepted_questions_with_reviewer_edits_count = (
+            stat.accepted_questions_with_reviewer_edits_count)
+        stats_model.first_contribution_date = stat.first_contribution_date
+        stats_model.last_contribution_date = stat.last_contribution_date
+
+    suggestion_models.QuestionReviewStatsModel.update_timestamps_multi(  # type: ignore[type-var]
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.QuestionReviewStatsModel.put_multi(  # type: ignore[type-var]
+        stats_models_to_update)
+
+
+def update_translation_contribution_stats_at_submission(
+    suggestion: suggestion_registry.BaseSuggestion,
+    user_id: str, target_id: str) -> None:
+    """Creates/updates TranslationContributionStatsModel model for
+    given translation submitter when a translation is submitted.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            submitted.
+        user_id: str. The ID of the corresponding submitter.
+        target_id: str. The ID of the target entity being suggested to.
+    """
+    topic_id = ''
+    exp_opportunity_dict = (
+        opportunity_services.get_exploration_opportunity_summaries_by_ids(
+            [target_id]))
```

Comment:
Could you create a function in `opportunity_services` that will fetch just one summary and return it directly?

Reply:
Done